crates/bevy_math/src/isometry.rs

```diff
@@ -0,0 +1,187 @@
+//! Isometry types for expressing rigid motions in two and three dimensions.
+//!
+//! These are often used to express the relative positions of two entities (e.g. primitive shapes).
+//! For example, in determining whether a sphere intersects a cube, one needs to know how the two are
+//! positioned relative to one another in addition to their sizes.
+//! If the two had absolute positions and orientations described by isometries `cube_iso` and `sphere_iso`,
+//! then `cube_iso.inverse() * sphere_iso` would describe the relative orientation, which is sufficient for
+//! answering this query.
+
+use crate::{Affine2, Affine3, Affine3A, Mat3, Mat3A, Quat, Rot2, Vec2, Vec3, Vec3A};
+use std::ops::Mul;
+
+/// An isometry in two dimensions.
+#[derive(Copy, Clone, Debug, PartialEq)]
```

Comment:
Should we have a `Default`, `Serialize` (behind the feature flag), and `Reflect` (also behind a feature flag)?